prime-router/src/main/kotlin/transport/SoapTransport.kt

```diff
@@ -0,0 +1,237 @@
+            // If Ktor fails to connect, or the server returns an error code, it is thrown
+            // as an exception higher up, which we catch and then track here. We do not need
+            // to worry about capturing and parsing out the return value from the response
+            // because Ktor treats errors as exceptions
+            val msg = "FAILED SOAP of inputReportId ${header.reportFile.reportId} to " +
+                "$soapTransportType (orgService = ${header.receiver.fullName})" +
+                ", Exception: ${t.localizedMessage}"
+            context.logger.severe(msg)
+            context.logger.severe(t.stackTraceToString())
+            actionHistory.setActionType(TaskAction.send_warning)
+            actionHistory.trackActionResult(msg)
+            RetryToken.allItems
+        }
+    }
+
+    /**
+     * Fetch the [SoapCredential] for a given [Receiver].
+     * @return the SOAP credential
+     */
+    internal fun lookupCredentials(receiver: Receiver): SoapCredential {
+        Preconditions.checkNotNull(receiver.transport)
+        Preconditions.checkArgument(receiver.transport is SoapTransportType)
+        val soapTransportInfo = receiver.transport as SoapTransportType
+
+        // if the transport definition has defined default
+        // credentials use them, otherwise go with the
+        // standard way by using the receiver full name
+        val credentialName = soapTransportInfo.credentialName ?: receiver.fullName
+
+        val credentialLabel = credentialName
+            .replace(".", "--")
+            .replace("_", "-")
+            .uppercase()
+        // Assumes credential will be cast as SftpCredential, if not return null, and thus the error case
+        return CredentialHelper.getCredentialService().fetchCredential(
+            credentialLabel, "SoapTransport", CredentialRequestReason.SOAP_UPLOAD
+        ) as? SoapCredential?
+            ?: error("Unable to find SOAP credentials for $credentialName connectionId($credentialLabel)")
+    }
+
+    companion object {
+        /** A default value for the timeouts to connect and send messages */
+        private const val TIMEOUT = 50_000
+
+        /** Our default Http Client */
+        private fun createDefaultHttpClient(): HttpClient {
+            return HttpClient(Apache) {
+                // installs logging into the call to post to the server
+                install(Logging) {
+                    logger = io.ktor.client.features.logging.Logger.Companion.SIMPLE
+                    level = LogLevel.INFO
+                }
+                // configures the Apache client with our specified timeouts
+                engine {
+                    followRedirects = true
+                    socketTimeout = TIMEOUT
+                    connectTimeout = TIMEOUT
+                    connectionRequestTimeout = TIMEOUT
+                    customizeClient {
+                    }
+                }
+            }
+        }
+    }
+}
```

Comment:
I extracted out the creation of the HTTP client to a separate method so we could mock and test the SOAP transport and inject in a different engine.